Add Literate docs for advection-diffusion

jlchan opened the pull request:
This PR adds a Literate tutorial for advection-diffusion. Questions:
- Should we also add a literate tutorial for `CompressibleNavierStokesDiffusion2D`?
- Should the new functionality (such as the gradient/divergence boundary conditions) be documented elsewhere in another page too?

ranocha commented:
> Should we also add a literate tutorial for `CompressibleNavierStokesDiffusion2D`?

Maybe - but we could also just mention it for now in the new tutorial and point to the `examples` folder.

> Should the new functionality (such as the gradient/divergence boundary conditions) be documented elsewhere in another page too?

We are not really good at documenting. Right now, the hyperbolic BCs are also only documented in docstrings and one tutorial, so it's fine as it is.

ranocha commented on docs/make.jl:
```diff
@@ -39,6 +39,7 @@ files = [
     # Topic: equations
     "Adding a new scalar conservation law" => "adding_new_scalar_equations.jl",
     "Adding a non-conservative equation" => "adding_nonconservative_equation.jl",
+    "Adding parabolic terms" => "adding_parabolic_terms.jl",
```
I would remove the "adding" since we don't really add new functionality here, just combine existing equations. The other "adding" tutorials are about creating new equations for Trixi.jl.

ranocha commented on docs/literate/src/files/adding_parabolic_terms.jl:
```diff
@@ -0,0 +1,88 @@
+#src # Adding parabolic terms (advection-diffusion).
```
Did you check the output?

jlchan replied:
Yup; looks like the following
<img width="907" alt="Screen Shot 2022-08-25 at 6 58 36 AM" src="https://user-images.githubusercontent.com/1156048/186658424-8b65366d-035a-46ca-818a-5a7c2bf44990.png">

ranocha commented on docs/literate/src/files/adding_parabolic_terms.jl:
```diff
@@ -0,0 +1,88 @@
+equations_hyperbolic = LinearScalarAdvectionEquation2D(advection_velocity);
+
+# Next, we define the parabolic diffusion term. The constructor requires knowledge of
+# `equations_hyperbolic` to be passed in because the `LaplaceDiffusion2D` applies
```
```suggestion
# `equations_hyperbolic` to be passed in because the [`LaplaceDiffusion2D`](@ref) applies
```

ranocha commented on docs/literate/src/files/adding_parabolic_terms.jl:
```diff
@@ -0,0 +1,88 @@
+
+# ## Semidiscretizing and solving
+
+# To semidiscretize a hyperbolic-parabolic system, we create a `SemidiscretizationHyperbolicParabolic`.
+# This differs from a `SemidiscretizationHyperbolic` in that we pass in a `Tuple` containing both the
```
```suggestion
# To semidiscretize a hyperbolic-parabolic system, we create a [`SemidiscretizationHyperbolicParabolic`](@ref).
# This differs from a [`SemidiscretizationHyperbolic`](@ref) in that we pass in a `Tuple` containing both the
```

ranocha reviewed (commented):
Thanks! Looks good to me. I just have a question and some minor suggestions.

jlchan commented:
Thanks @ranocha! I've addressed your comments. Once I add some documentation of the Gradient/Divergence BCs, I think this will address the issue of docs in https://github.com/trixi-framework/Trixi.jl/issues/1147 (I'll do this in a separate PR though)

ranocha reviewed (approved):
Thanks!
